### PR TITLE
[core] don't broadcast node failures to drivers and instead log the error

### PR DESCRIPTION
**Pull request by AmeerHajAli**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — COMMENTED** [review]
Actually can we make the message INFO? If it is error, it is probably still propagated to the driver

also cc @ericl can you give the approval?

**Comment by AmeerHajAli** [comment]
We might go with the rpc solution as well.

File: src/ray/gcs/gcs_server/gcs_node_manager.cc [hunk 1]
```diff
@@ -149,24 +149,13 @@ std::shared_ptr<rpc::GcsNodeInfo> GcsNodeManager::RemoveNode(
...
-                       "raylet crashes unexpectedly or has lagging heartbeats.";
-      auto error_data_ptr =
-          gcs::CreateErrorTableData(type, error_message.str(), current_time_ms());
-      RAY_EVENT(ERROR, EL_RAY_NODE_REMOVED)
-              .WithField("node_id", node_id.Hex())
-              .WithField("ip", removed_node->node_manager_address())
-          << error_message.str();
-      RAY_CHECK_OK(gcs_pub_sub_->Publish(ERROR_INFO_CHANNEL, node_id.Hex(),
-                                         error_data_ptr->SerializeAsString(), nullptr));
+      RAY_LOG(ERROR) << "The node with node id: " << node_id
```
Comment (ericl) [inline src/ray/gcs/gcs_server/gcs_node_manager.cc:154]: INFO or warn

Comment (AmeerHajAli) [inline src/ray/gcs/gcs_server/gcs_node_manager.cc:154]:
```suggestion
      RAY_LOG(WARNING) << "The node with node id: " << node_id
```

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
Sounds like we decided not to merge this.

**Pull request closed without merging**